src/Workspaces/Remote/ServiceHub/Host/AssetProvider.cs

```diff
@@ -139,41 +153,90 @@ async ValueTask SynchronizeProjectAssetsWorkerAsync()
             // get children of project checksum objects at once
             using var _ = PooledHashSet<Checksum>.GetInstance(out var checksums);
 
-            checksums.Add(projectChecksums.Info);
-            checksums.Add(projectChecksums.CompilationOptions);
-            checksums.Add(projectChecksums.ParseOptions);
-            AddAll(checksums, projectChecksums.ProjectReferences);
-            AddAll(checksums, projectChecksums.MetadataReferences);
-            AddAll(checksums, projectChecksums.AnalyzerReferences);
-            AddAll(checksums, projectChecksums.Documents.Checksums);
-            AddAll(checksums, projectChecksums.AdditionalDocuments.Checksums);
-            AddAll(checksums, projectChecksums.AnalyzerConfigDocuments.Checksums);
+            {
```

Comment:
there's a lot of duplication that happens in the new code.  the followup moves those into helpers that dedupe the pattern.

Reply:
the followup also makes us do this work in parallel